### PR TITLE
feat: support .gjs in Ember PackageGraphs

### PR DESCRIPTION
**Pull request by wondersloth**
This PR addresses #606 

Adds support for `.gjs` files to Ember PackageGraphs.

- Includes `.gjs` file extension for `EmberAppPackageGraph` and `EmberAddonPackageGraph` resolveOptions passed to dependency-cruiser.
- Added test for `EmberAddonPackage` due to bug found where resolve alias for ember-addon paths starting with moduleName where not mapping back to the addon directory when regressing against internal addon.
- Preprocess/transform `.gjs` files to remove `<template>` tags and use something more es parse-able.

### Testing Done
- Added tests
- Regressed against internal addon `voyager-web/packages/addons/image-view-model` which has `.gjs` files. Produces graph edges correctly. 

### TODO
- [x] Add service discovery support for .gjs files.
- [x] Drop `@glimmer/babel-preset` as we don't need to transform **`hbs`**. 
- [X] Figure out how to infer `@glimmer/syntax` dependency from target package's node_modules so we can use the version related to the ember app.

### ~Concerns/Alternatives~ 
**This is no longer a concern because I removed usage of `futureCruise` api that was experimental/unstable.**

- The `futureCruise` method from `dependency-cruiser` API is experimental/unstable.
- Not currently a problem, but a potential future risk

#### Drop futureCruise API, no need for `@glimmer/babel-preset`
- This may sound odd, but we don't need `@glimmer/babel-preset` to add `.gjs` files to the graph using `dependency-cruiser`
- I was able to implement this without using `@glimmer/babel-preset` because ember  `<template>` tag semantics get seen as JSX TemplateExpressions.
- When `dependency-cruiser` parses these files, it thinks they're JSX like source files and manages to get the import declarations perfectly fine.
- First couple of commits to this PR have just that. ~Last commit of this PR is the refactor to use `futureCruise`.~ We don't use futureCruiser at all. Just plain old cruise. 

#### Find another graph solution
- Some other OTS graph creation tool.
- Pros:
  - Maybe more stable API?
- Cons:
  - Have to find and adapt another solution.
  - Uncertain if it will provide same file level data.
  - Uncertain if it will meet or exceed current perf.

#### Create our own dependency graph solution
- Expand on the naive `swc` es6 entrypoint walker I had before.
- Pros: We control our own crawl tools. Maybe faster.
- Cons:
  - Have to re-invent the wheel.
  - This is a solved problem, by mostly all package managers.
 

#### Explore and emit the resolved graph data from rollup when passed an entrypoint

Implementation:
- fast glob all entrypoints in a project `.` directories and top level files.
- Then iterate with rollup

Pros:
- If it works, a way to customize resolutions for different module environments.
- Well rationalized API.

Cons:
- Perf is uncertain.
- Uncertain of success
- Have to create a new solution.

### CODE REVIEW

**Comment by wondersloth** [comment]
Summary of work:

- Did not use `@glimmerx/babel-preset` or `@glimmer/babel-preset` as I want to produce some module source format that can be parsed by SWC.
- If I were to use babel to **transform** the source file using:
```
  const { code } = transformSync(output, {
    filename,
    targets: {
      node: 'current',
    },
    presets: [['@glimmerx/babel-preset', { __loadPlugins: true }]],
  });
```

It produces output that is not what we need for passing to swc to parse out the service usage. 

TL;DR I wrote the swc parser to work against es6 source, so the output from babel transform will lose that fidelity to parse the AST for service usage.

**Comment by wondersloth** [comment]
Note: Prior concerns about alternative dependency graph tools, are minor because we're no longer using experimental API from dependency cruiser.

**Review by chadhietala — APPROVED** [review]
This looks great @wondersloth.

**Review by lynchbomb — APPROVED** [review]
